gp-includes/routes/glossary.php

```diff
@@ -15,8 +15,19 @@
 class GP_Route_Glossary extends GP_Route_Main {
 
 	public function new_get() {
-		$glossary                     = new GP_Glossary();
-		$glossary->translation_set_id = gp_get( 'translation_set_id' );
+		$glossary = new GP_Glossary();
+
+		$translation_set_id = gp_get( 'translation_set_id' );
+
+		// Make sure 'translation_set_id' is a string and convert to int ID. Defaults to null.
+		$translation_set_id = is_string( $translation_set_id ) && intval( $translation_set_id ) ? intval( $translation_set_id ) : null;
```

Comment:
I have used some different GET parameters, and we have a problem with the strings starting with a number. E.g., if the pamater is `1234www`, the `intval` function returns `1234`, so we need to change it:

- `/glotpress/glossaries/-new/?translation_set_id=fasdf` 
- `/glotpress/glossaries/-new/?translation_set_id=123`
- `/glotpress/glossaries/-new/?translation_set_id=1s`
- `/glotpress/glossaries/-new/?translation_set_id[]=1`
- `/glotpress/glossaries/-new/?translation_set_id[1]=1ss`
- `/glotpress/glossaries/-new/?translation_set_id=1234www`
```suggestion
		$translation_set_id = filter_var( $translation_set_id, FILTER_VALIDATE_INT ) !== false ? intval( $translation_set_id ) : null;
```

Reply:
As we've came to the conclusion in Slack, I'm making it more simpler with `is_numeric()`.